slack_sdk/models/dialogs/__init__.py

```diff
@@ -414,14 +414,14 @@ def __init__(
             name=name,
             label=label,
             value=value,
-            optional=optional,
+            optional=optional,  # type: ignore[arg-type]
             placeholder=placeholder,
         )
         self.min_query_length = min_query_length
 
 
 class DialogBuilder(JsonObject):
-    attributes = {}  # no attributes because to_dict has unique implementation
+    attributes = set()  # no attributes because to_dict has unique implementation
```

Comment:
It seems this property is unused, so changing the type should not affect the behavior. That said, changing from dict to set object may not make sense. This legacy component won't be enhanced anymore but using a set object might be a cause of future bugs?

Reply:
Good point 💯 it seems like the `attribute` property is defined as a `set` in the inherited class (`JsonObject`), I'm assuming this object overrides this and `mypy` does not like it. I've reverted the changes and added an ignore comment for now

We can address these ignore comments later or if they become relevant